rest-server/core/src/main/scala/com/actionml/core/spark/SparkMongoSupport.scala

```diff
@@ -18,15 +18,17 @@
 package com.actionml.core.spark
 
 import com.mongodb.spark.MongoSpark
-import com.mongodb.spark.rdd.MongoRDD
 import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
 import org.bson.Document
 
 
-trait SparkMongoSupport {
+trait SparkStorageSupport[T] {
+  def createRdd(sc: SparkContext): RDD[T]
+}
```

Comment:
ok but ideally wouldn't you pass in a DAO with a filter and ordering? Then the Abstract `SparkStorageSupport` would have enough abstract DAO params to execute the implementation of `createRdd`, right?

BTW some naming conventions: 1) I renamed Storage to Store 2) shouldn't this be `readRdd` since we'll eventually have `writeRdd` at least in the Elasticsearch case and probably for any Store.

Reply:
> ok but ideally wouldn't you pass in a DAO with a filter and ordering? Then the Abstract SparkStorageSupport would have enough abstract DAO params to execute the implementation of createRdd, right?

I see no support for filtering and ordering inside mongo-spark lib. So, I suppose, we shouldn't add those params to our API since our only backend doesn't support it.

Fixed naming.

Reply:
hmm, so the only thing it does is read the whole collection into an RDD? Then we filter in Spark. It is common enough that we can add that the our abstract API. You may want to curry the `sc` if you do an `rdd.filter`